### PR TITLE
Let CMake detect toolkits/compilers by default

### PR DESCRIPTION
**Pull request by tbennun**
Addresses issue #59 and fixes cross-platform (i.e. Windows) builds without changing environment variables.

Still missing FPGA part @definelicht.

### CODE REVIEW

File: dace/config_schema.yml [hunk 1]
```diff
@@ -132,7 +112,7 @@ required:
 
             allow_shadowing:
                 type: str
-                default: false 
+                default: true
```
Comment (definelicht) [inline dace/config_schema.yml]: Is this related, or did it just sneak in here? :-)

**Comment by definelicht** [comment]
Nice job Tal, should've done this a long time ago. I'll do the FPGA part.

File: dace/config_schema.yml (hunk 1 above)
Comment (tbennun) [reply dace/config_schema.yml]: Snuck in, can revert but I think it should be in master until the new codegen appears
